src/algorithm.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+// Pragma once is supported on all modern compilers
+// TODO: decide whether to use include guards anyway
+
+#include <algorithm>
+
+//--------------------------------------------------------------------------------------------------
+// Container-wide versions of std algoithms.
+//--------------------------------------------------------------------------------------------------
+
+template <typename Container, typename Predicate>
+inline bool any_of(Container&& c, Predicate p) {
+    using std::begin; //ADL
```

Comment:
Yes :smile: Argument-dependent Lookup aka Koenig lookup aka gonna ruin your day sometime in the future.

If you're not being sarcastic, what would you have assumed that purpose of those using statements was without the comment? I'm genuinely curious.